crates/project/src/project.rs

```diff
@@ -496,6 +496,7 @@ impl ProjectEntryId {
 pub enum FormatTrigger {
     Save,
     Manual,
+    SaveWithoutFormatting,
```

Comment:
Is it possible to get rid of this and handle the checks in here one layer up? It seems a bit weird to me that we have a trigger-to-format that includes "without formatting" :) Meaning: why have a noop trigger when we could instead not trigger?

Reply:
> It seems like we should change this behavior so `format_on_save: false` and save without formatting don't format whitespace.

I think that makes sense. Another thought: maybe if we make the behavior more explicit (i.e. have something like `StripWhitespace` somewhere) things can become clearer in the code?